google-cloud-clients/google-cloud-core/src/main/java/com/google/cloud/testing/BaseEmulatorHelper.java

```diff
@@ -366,7 +370,15 @@ public boolean isAvailable() {
 
     @Override
     public void start() throws IOException {
-      Path emulatorPath = downloadEmulator();
+      ExceptionHandler retryOnAnythingExceptionHandler = ExceptionHandler.newBuilder().retryOn(Exception.class).build();
+
+      Path emulatorPath = RetryHelper.runWithRetries(new Callable<Path>() {
+          @Override
+          public Path call() throws IOException {
+              return downloadEmulator();
```

Comment:
Can we get into some weird state where it will retry a known failed state? Perhaps the download is corrupt and we retry but the file exists and continues to fail on the unzip?

Reply:
Ok